lib/ansible/modules/git.py

```diff
@@ -824,13 +824,14 @@ def get_head_branch(git_path, module, dest, remote, bare=False):
     """
     try:
         repo_path = get_repo_path(dest, bare)
-    except (IOError, ValueError) as err:
+    except (OSError, ValueError) as ex:
         # No repo path found
         # ``.git`` file does not have a valid format for detached Git dir.
         module.fail_json(
             msg='Current repo does not have a valid reference to a '
             'separate Git dir or it refers to the invalid path',
-            details=to_text(err),
+            details=to_text(ex),
```

Comment:
No need for `to_text` on an exception, `str` works:

```suggestion
            details=str(ex),
```